Timeouts and timing changes

tom-acco opened the pull request:
Started adding some timeouts for following and unfollowing users, if it fails 3 times it'll skip and move on... Haven't tested this properly yet and I still need to add some more for login fails and such.
Changed some of the sleep timings
- Utilising randint(x,y) for the timings between actions that 500PX may see. This is to give it more of a "human" like look... hopefully?

tom-acco committed:
Timeouts and timing changes

Started adding some timeouts for following and unfollowing users, if it fails 3 times it'll skip and move on... Haven't tested this properly yet and I still need to add some more for login fails and such.
Changed some of the sleep timings
- Utilising randint(x,y) for the timings between actions that 500PX may see. This is to give it more of a "human" like look... hopefully?

skuseme commented:
Ran this last night and it ran fine.

tom-acco committed:
Timeout fixes & updates

Fixed the followUser timeout - had incorrect indentation
Added timeout for retrieving login page - if you get blocked for too many requests 500px won't let you login.
- Will attempt to retrieve login page 3 times, if it fails 3 times it will wait an hour and try again.

Kenshii commented:
I'd prefer to keep the timeouts as-is, because when I attempted to lower them several times in the past, it only resulted in the site banning me.

skuseme commented:
That's understandable. I'll increase them but keep them at a small randint.

tom-acco committed:
Lengthen timeouts

Timeouts shorter than 20 seconds have been known to ban users. Increased the random integer to 20-30
